proxy/sources/catbox.py

```diff
@@ -0,0 +1,116 @@
+from datetime import datetime
+from typing import List, Optional, Dict
+
+from django.shortcuts import redirect
+from django.urls import re_path
+
+from proxy.source import ProxySource, SeriesPage, ChapterAPI, SeriesAPI, api_cache, post_wrapper
+
+
+class Catbox(ProxySource):
+    def get_reader_prefix(self) -> str:
+        return "catbox"
+
+    def shortcut_instantiator(self) -> List[re_path]:
+        def handler(_, album_hash):
+            slug = f"reader-{self.get_reader_prefix()}-chapter-page"
+            return redirect(slug, album_hash, "1", "1")
+
+        return [re_path(r"^c/(?P<album_hash>\w+)/$", handler)]
+
+    @api_cache(prefix="catbox_api_dt", time=300)
+    def catbox_common(self, meta_id: str) -> Optional[Dict]:
+        resp = post_wrapper(f"https://catbox.moe/user/api.php", data={"reqtype": "getalbum", "short": meta_id})
+
+        if resp.status_code != 200:
+            return None
+
+        json = resp.json()
+
+        if json["status"] != 200 or not json["success"]:
+            return None
```

Comment:
Huh, this was an incredibly difficult API to track down. There's virtually no documentation, and the first reference I found was in the [node library](https://github.com/tentaclenotsoft/node-catbox/blob/ac1844f8307a6b645ce64dd2eb40592f3ef684cd/lib/structures/Catbox.js#L53-L68) linked on [their tools doc](https://catbox.moe/tools.php).

Anyway, this is a strange API:
* The returned content-type is `text/html`, despite the body being a JSON body
* The JSON includes a `status` and `success` field that you're checking -- it's unclear whether the HTTP status code is correctly used here
* Creating an album requires a dance of uploading the individual files, then creating a collection with the shortcodes after you've uploaded all your files
* There's no validation for the album creation -- you can pass in any arbitrary string of images

```bash
# Create an "album"
curl -F "reqtype=createalbum" -F 'title=title' -F 'desc=desc' -F 'files="a b c d e f"' https://catbox.moe/user/api.php

# From the shortcode above
curl -F "reqtype=getalbum" -F "short=0j1hb2" https://catbox.moe/user/api.php
```

Weird.